### DISCUSSION TITLE
JDK 11 HttpSession ServletUtilities

### DISCUSSION THREAD

**Discussion by scott-goodell** [jfree/jfreechart#371] [Q&A]
I have recently been working on upgrading my web server to Tomcat 10 and JDK 11 and ran across the following issue with jfreechart.

The method saveChartAsPNG(JFreeChart, int, int, ChartRenderingInfo, HttpSession) from the type ServletUtilities refers to the missing type HttpSession

With JDK 11 HttpSession was moved from javax to jakarta.  I ended up doing the following to get around the issue.

Download  jfreechart-1.5.x.zip  from github and open in 7zip and extract ChartDeleter.java, ServletUtilities.java, DisplayChart.java
changed javax to jakarta in each java program
javac -cp "..." org\jfree\chart\servlet\ChartDeleter.java
jar -uf jfreechart-1.5.3.jar org\jfree\chart\servlet\ChartDeleter.class
javac -cp "..." org\jfree\chart\servlet\ServletUtilities.java
jar -uf jfreechart-1.5.3.jar org\jfree\chart\servlet\ServletUtilities.class
javac -cp "..." org\jfree\chart\servlet\DisplayChart.java
jar -uf jfreechart-1.5.3.jar org\jfree\chart\servlet\DisplayChart.class

I'm no Java expert but was able to get this resolved but if someone has other suggestions I am open to hearing about it.

Thanks

**Comment by trashgod** [comment]
You can branch and build from the desired tag like [this](https://stackoverflow.com/a/69653678/230513). I'm not sure it's better, but it may help with configuration management.
